src/python/pants/engine/fs.py

```diff
@@ -157,6 +157,14 @@ def is_empty(self):
     return self == EMPTY_SNAPSHOT
 
 
+@dataclass(frozen=True)
+class SnapshotSubset:
+  """A request to create a subset of a snapshot."""
+  directory_digest: Digest
+  include_files: Tuple[str, ...]
+  include_dirs: Tuple[str, ...]
```

Comment:
Consider giving these both a default value of `()`. It seems likely that a user will want to include files and not care about directories, and vice versa.

Reply:
See my comment on #8986 about this.